pkg/sync/sync_context.go

```diff
@@ -563,8 +563,12 @@ func (sc *syncContext) getSyncTasks() (_ syncTasks, successful bool) {
 
 		if sc.applyOutOfSyncOnly {
 			if modified, ok := sc.modificationResult[k]; !modified && ok {
-				sc.log.WithValues("resource key", k).V(1).Info("Skipping as resource was not modified")
-				continue
+				if sc.prune && resource.Target == nil {
```

Comment:
I might be wrong but I think we need to check for `resource.Target == nil || resource.Live == nil`. In both cases resource is not in sync.

Reply:
if `resource.Live == nil` then modified will be true. In that case, this resource won't be skip. Hence, we don't need to check if `resource.Live == nil`.